anax-core/src/main/java/org/anax/framework/controllers/AppiumDriverWebController.java

```diff
@@ -25,13 +26,17 @@
 @Slf4j
 public class AppiumDriverWebController implements WebController{
 
+    /** Reference to the AnaxDriver which holds the construction method of This Controller **/
+    private final AnaxDriver anaxDriver;
+
     private WebDriver driver;
 
     /** The Constant THREAD_SLEEP. */
     private final long defaultWaitSeconds;
 
-    public AppiumDriverWebController(WebDriver webDriver, long defaultWaitSeconds) {
+    public AppiumDriverWebController(WebDriver webDriver, AnaxDriver anaxDriver, long defaultWaitSeconds) {
         this.driver = webDriver;
+        this.anaxDriver = anaxDriver;
```

Comment:
replace `this.driver = webDriver` with `this.driver = anaxDriver.getWebDriver()` - much cleaner after you simplify the signature (see first comment)